client/src/components/ActionColumn.js

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { Table } from "semantic-ui-react";
+import { Link } from "react-router-dom";
+
+const ActionColumn = props => {
+  const { id, status, translatorName, userName, userRole } = props;
+  if (userRole === "User") {
+    return (
+      <Table.Cell>
+        <span>
+          <Link to={`/document/${id}`}>View</Link>
+        </span>{" "}
```

Comment:
You are using this code for the View link 3 times in this component, so you could extract this bit in a another component (you can still define it here in the same file) and use that new component instead, to avoid duplication.

Reply:
Changed